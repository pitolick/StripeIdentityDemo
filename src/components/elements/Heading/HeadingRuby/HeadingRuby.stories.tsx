import React from 'react'
import { Meta, StoryObj } from '@storybook/react'
import { HeadingRuby } from './HeadingRuby'

const meta: Meta<typeof HeadingRuby> = {
  title: 'elements/Heading/HeadingRuby',
  component: HeadingRuby,
  tags: ['autodocs'],
}

export default meta
type Story = StoryObj<typeof HeadingRuby>

/**
 * デフォルト
 */
export const Default: Story = {
  args: {
    children: '見出し',
    ruby: 'サブタイトル',
  },
}

/**
 * h1
 */
export const H1: Story = {
  args: {
    children: '見出し',
    ruby: 'サブタイトル',
    tag: 'h1',
  },
}

/**
 * h2
 */
export const H2: Story = {
  args: {
    children: '見出し',
    ruby: 'サブタイトル',
    tag: 'h2',
  },
}

/**
 * h3
 */
export const H3: Story = {
  args: {
    children: '見出し',
    ruby: 'サブタイトル',
    tag: 'h3',
  },
}

/**
 * h4
 */
export const H4: Story = {
  args: {
    children: '見出し',
    ruby: 'サブタイトル',
    tag: 'h4',
  },
}

/**
 * h5
 */
export const H5: Story = {
  args: {
    children: '見出し',
    ruby: 'サブタイトル',
    tag: 'h5',
  },
}

/**
 * h6
 */
export const H6: Story = {
  args: {
    children: '見出し',
    ruby: 'サブタイトル',
    tag: 'h6',
  },
}

/**
 * カスタムクラス
 */
export const CustomClass: Story = {
  args: {
    children: '見出し',
    ruby: 'サブタイトル',
    className: 'text-red',
  },
}

/**
 * 子要素
 */
export const Children: Story = {
  args: {
    children: (
      <>
        <span className="text-4xl">これは見出しです。</span>
        <br />
        <span className="text-xl">これはサブタイトルです。</span>
      </>
    ),
    ruby: 'サブタイトル',
  },
}
