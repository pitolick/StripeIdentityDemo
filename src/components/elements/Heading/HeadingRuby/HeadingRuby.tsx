/**
 * ルビ付きの見出し
 * @module HeadingRuby
 * @example
 * <HeadingRuby>見出し</HeadingRuby>
 */
import React from 'react'

/**
 * ルビ付きの見出しコンポーネントのプロパティ
 */
interface HeadingRubyProps {
  /**
   * 子要素
   */
  children: React.ReactNode
  /**
   * ルビ
   */
  ruby: string
  /**
   * タグ名
   */
  tag?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
  /**
   * class名
   */
  className?: React.HTMLAttributes<HTMLHeadingElement>['className']
}

/**
 * ルビ付きの見出しコンポーネント
 */
export const HeadingRuby: React.FC<HeadingRubyProps> = ({ children, ruby, tag: CustomTag = 'h2', className }) => {
  return (
    <>
      <CustomTag className={`${className}`}>
        <span className="block text-base text-gray-900">{ruby}</span>
        <span className="text-3xl font-bold">{children}</span>
      </CustomTag>
    </>
  )
}
