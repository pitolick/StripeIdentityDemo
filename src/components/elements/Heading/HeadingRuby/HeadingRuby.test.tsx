import React from 'react'
import { render } from '@testing-library/react'
import { HeadingRuby } from './HeadingRuby'

describe('HeadingRuby', () => {
  it('子要素が正しく表示されること', () => {
    const { container } = render(<HeadingRuby ruby="見出し">見出し</HeadingRuby>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h1タグが正しく設定されること', () => {
    const { container } = render(
      <HeadingRuby ruby="見出し" tag="h1">
        見出し
      </HeadingRuby>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h2タグが正しく設定されること', () => {
    const { container } = render(
      <HeadingRuby ruby="見出し" tag="h2">
        見出し
      </HeadingRuby>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h3タグが正しく設定されること', () => {
    const { container } = render(
      <HeadingRuby ruby="見出し" tag="h3">
        見出し
      </HeadingRuby>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h4タグが正しく設定されること', () => {
    const { container } = render(
      <HeadingRuby ruby="見出し" tag="h4">
        見出し
      </HeadingRuby>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h5タグが正しく設定されること', () => {
    const { container } = render(
      <HeadingRuby ruby="見出し" tag="h5">
        見出し
      </HeadingRuby>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h6タグが正しく設定されること', () => {
    const { container } = render(
      <HeadingRuby ruby="見出し" tag="h6">
        見出し
      </HeadingRuby>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('class名が正しく設定されること', () => {
    const { container } = render(
      <HeadingRuby ruby="見出し" className="text-red">
        見出し
      </HeadingRuby>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
