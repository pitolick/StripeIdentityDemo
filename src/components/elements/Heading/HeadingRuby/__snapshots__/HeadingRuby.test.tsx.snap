// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingRuby class名が正しく設定されること 1`] = `
<h2
  class="text-red"
>
  <span
    class="block text-base text-gray-900"
  >
    見出し
  </span>
  <span
    class="text-3xl font-bold"
  >
    見出し
  </span>
</h2>
`;

exports[`HeadingRuby h1タグが正しく設定されること 1`] = `
<h1
  class="undefined"
>
  <span
    class="block text-base text-gray-900"
  >
    見出し
  </span>
  <span
    class="text-3xl font-bold"
  >
    見出し
  </span>
</h1>
`;

exports[`HeadingRuby h2タグが正しく設定されること 1`] = `
<h2
  class="undefined"
>
  <span
    class="block text-base text-gray-900"
  >
    見出し
  </span>
  <span
    class="text-3xl font-bold"
  >
    見出し
  </span>
</h2>
`;

exports[`HeadingRuby h3タグが正しく設定されること 1`] = `
<h3
  class="undefined"
>
  <span
    class="block text-base text-gray-900"
  >
    見出し
  </span>
  <span
    class="text-3xl font-bold"
  >
    見出し
  </span>
</h3>
`;

exports[`HeadingRuby h4タグが正しく設定されること 1`] = `
<h4
  class="undefined"
>
  <span
    class="block text-base text-gray-900"
  >
    見出し
  </span>
  <span
    class="text-3xl font-bold"
  >
    見出し
  </span>
</h4>
`;

exports[`HeadingRuby h5タグが正しく設定されること 1`] = `
<h5
  class="undefined"
>
  <span
    class="block text-base text-gray-900"
  >
    見出し
  </span>
  <span
    class="text-3xl font-bold"
  >
    見出し
  </span>
</h5>
`;

exports[`HeadingRuby h6タグが正しく設定されること 1`] = `
<h6
  class="undefined"
>
  <span
    class="block text-base text-gray-900"
  >
    見出し
  </span>
  <span
    class="text-3xl font-bold"
  >
    見出し
  </span>
</h6>
`;

exports[`HeadingRuby 子要素が正しく表示されること 1`] = `
<h2
  class="undefined"
>
  <span
    class="block text-base text-gray-900"
  >
    見出し
  </span>
  <span
    class="text-3xl font-bold"
  >
    見出し
  </span>
</h2>
`;
